src/main/java/org/truffleruby/parser/lexer/RubyLexer.java

```diff
@@ -3084,7 +3084,7 @@ public void setValue(Object yaccValue) {
     }
 
     protected boolean strncmp(Rope one, Rope two, int length) {
-        if (one.byteLength() < length || two.byteLength() < length) {
+        if (one.byteLength() != two.byteLength() && (one.byteLength() < length || two.byteLength() < length)) {
```

Comment:
Does it make sense to call `strncmp` when length is bigger than one of the two Ropes?
Or should this be some kind of `assert`?

What should happen if `one.byteLength() == two.byteLength()` but `length > one.byteLength()`?

Reply:
I'm going to look at cleaning this up in another PR. I should've just left the short-circuit check alone and can back that commit out if you'd prefer. But it struck me as being blatantly wrong.

Reply:
I think in the case I described above, the `assert` in `ArrayUtils.memcmp` is going to trigger.
And I feel there is a semantic problem here, we should only `strncmp` if length is smaller than both Ropes' byteLengths.
Feel free to fix in another PR though.

Reply:
It's code we inherited and it seems to work. The length is chosen as the length of one of the ropes in all usages, so at least one of the ropes is never smaller than length. But, yes, let's defer to another PR.